### ISSUE TITLE
Change Request: allow `default` as known key for interception between commonjs and ESM

### ISSUE THREAD

**Issue by JounQin** [eslint/eslint#19703]
### ESLint version

v9.26.0

### What problem do you want to solve?

`eslint-config-prettier` is commonjs only + `exports.rules`/`exports.name`, so `import * as eslintConfigPrettier from 'eslint-config-prettier'` should be used when using `eslint.config.mjs`, but node will add an extra `default` property on `eslintConfigPrettier` in this case.

```js
import * as eslintConfigPrettier from 'eslint-config-prettier'

console.log(Object.keys(eslintConfigPrettier)) // [ 'default', 'rules' ]

import * as eslintConfigPrettierFlat from 'eslint-config-prettier/flat'

console.log(Object.keys(eslintConfigPrettierFlat)) // [ 'default', 'name', 'rules' ]
```

But those can not be used directly as ESLint flat config due to the `default` property:

```log
Oops! Something went wrong! :(

ESLint: 9.26.0

ConfigError: Config (unnamed): Unexpected key "default" found.
```

### What do you think is the correct solution?

1. Ignore `default` key
2. Ignore all unknown keys, I don't get why we're vailidating the unknown keys actually, for example, we had to add an extra `/flat` entry only for adding a new `name` property for `flat` config which is unavailable in `eslintrc`, this is a bit ridiculous to myself: https://github.com/prettier/eslint-config-prettier/pull/309

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
> `eslint-config-prettier` is commonjs only + `exports.rules`/`exports.name`, so `import * as eslintConfigPrettier from 'eslint-config-prettier'` should be used when using `eslint.config.mjs`, but node will add an extra `default` property on `eslintConfigPrettier` in this case.
> 
> ```js
> import * as eslintConfigPrettier from 'eslint-config-prettier'
> 
> console.log(Object.keys(eslintConfigPrettier)) // [ 'default', 'rules' ]
> 
> import * as eslintConfigPrettierFlat from 'eslint-config-prettier/flat'
> 
> console.log(Object.keys(eslintConfigPrettierFlat)) // [ 'default', 'name', 'rules' ]
> ```

General recommendation for importing commonjs modules in ESM is to use default imports only.

So the usage should be:

```js
import eslintConfigPrettierFlat from 'eslint-config-prettier/flat'
```

Can you check if this works as expected in a config file?

> Ignore all unknown keys, I don't get why we're vailidating the unknown keys actually

Throwing an error on unknown keys is helpful to catch mistakes in the config, e.g., typos.

**Comment by mdjermanovic** [comment]
> > General recommendation for importing commonjs modules in ESM is to use default imports only.
> 
> Sorry I forgot to mention [prettier/eslint-config-prettier#324](https://github.com/prettier/eslint-config-prettier/issues/324), it conflicts with rule [`import/default`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/default.md), we have to [do this](https://github.com/prettier/eslint-config-prettier/pull/325) as workaround.

That might be a bug in the types and/or the rule. The recommended way to import a CommonJS module in ESM should be the default import, as already presented in the docs for eslint-config-prettier: https://github.com/prettier/eslint-config-prettier?tab=readme-ov-file#installation

> ```js
> import someConfig from "some-other-config-you-use";
> // Note the `/flat` suffix here, the difference from default entry is that
> // `/flat` added `name` property to the exported object to improve
> // [config-inspector](https://eslint.org/blog/2024/04/eslint-config-inspector/) experience.
> import eslintConfigPrettier from "eslint-config-prettier/flat";
> 
> export default [
>   someConfig,
>   eslintConfigPrettier,
> ];
> ```

Also, the module logically doesn't export `name` and `rules` as separate entities, but a config object with properties `name` and `rules`.

Given that the validation for unknown properties exists since eslintrc, and that it helps finding errors in the config and allows for introducing new config properties in the future, I don't think we'll make any changes here.

**Comment by JounQin** [comment]
@aladdin-add @nzakas Do you share same vision with @mdjermanovic?

**Comment by aladdin-add** [comment]
yes, I agreed - for max compat, it's recommended always using the default imports.
